src/main/java/org/apache/commons/text/WordUtils.java

```diff
@@ -739,4 +739,70 @@ private static boolean isDelimiter(final char ch, final char[] delimiters) {
         return false;
     }
 
-}
+    //-----------------------------------------------------------------------
+    /**
+     * Abbreviates a string nicely.
+     *
+     * This method searches for the first space after the lower limit and abbreviates
+     * the String there. It will also append any String passed as a parameter
+     * to the end of the String. The upper limit can be specified to forcibly
+     * abbreviate a String.
+     *
+     * @param str         the string to be abbreviated. If null is passed, null is returned.
+     *                    If the empty String is passed, the empty string is returned.
+     * @param lower       the lower limit.
+     * @param upper       the upper limit; specify -1 if no limit is desired.
+     *                    If the upper limit is lower than the lower limit, it will be
+     *                    adjusted to be the same as the lower limit.
+     * @param appendToEnd String to be appended to the end of the abbreviated string.
+     *                    This is appended ONLY if the string was indeed abbreviated.
+     *                    The append does not count towards the lower or upper limits.
+     * @return the abbreviated String.
+     */
+    public static String abbreviate(String str, int lower, int upper, String appendToEnd) {
+
+        if (StringUtils.isEmpty(str)) {
+            return str;
+        }
+
+        // throw IllegalArgumentException if upper limit is less than -1 which voids contact.
+        if (upper < -1) {
+            throw new IllegalArgumentException("upper value cannot be less than -1");
+        }
+
+        // throw IllegalArgumentException if upper value is less than lower value.
+        if (upper < lower && upper != -1) {
+            throw new IllegalArgumentException("upper value is less than lower value");
+        }
```

Comment:
I have moved validation on method start but cant find appropriate Validation method to use the way we are doing validations, also i think code here is easy to understand anyway.